### PR TITLE
Fix from_features issue resulting from dropped rows

### PR DESCRIPTION
**Pull request by gbrener**
Addresses issue #27. This solution is not expected to be performant, but it gets the job done - effectively by padding the data with `NaN` values in the coordinate locations necessary for the reshaping to work.

### CODE REVIEW

File: xarray_filters/reshape.py [hunk 1]
```diff
@@ -238,8 +238,23 @@ def from_features(arr, axis=0):
...
     dset = OrderedDict()
+    n_elts_needed = np.prod(shp)
     for j in range(simple_np_arr.size):
-        val = arr[:, j].values.reshape(shp)
+        arr_val = arr[:, j].values
+        n_arr_vals = len(arr_val)
+        assert n_arr_vals <= n_elts_needed
+        if n_arr_vals < n_elts_needed:
+            val = np.full(shp, np.nan)
+            feature_names = list(arr.indexes.keys())
```
Comment (PeterDSteinberg) [inline xarray_filters/reshape.py]: This looks good to me (line 243 to 257 as written now), but I think we can assume the MultiIndex is on the row axis or else the row axis is a simple index.  If we encounter a simple index on the row axis, and we have a single DataArray that is 2D (typically called `.features`), then we the reshaping doesn't happen and we just break out the columns of `.features` into separate 1D `DataArray`s.

**Comment by PeterDSteinberg** [comment]
One other thing to test @gbrener .  
 * [ ] Make an `MLDataset` that only as 1-dimensional `DataArray`s as `layers`, then call `to_features` on that, then drop some rows, then to `from_features` and make sure the `NaN`/`MultiIndex`-related logic in this PR is okay in that situation.

File: xarray_filters/reshape.py (hunk 1 above)
Comment (gbrener) [reply xarray_filters/reshape.py]: @PeterDSteinberg , so would removing lines 251 and 252 - and changing line 248 to what's below - be preferred?
```
feature_names = [arr.dims[0]]
```

Comment (PeterDSteinberg) [reply xarray_filters/reshape.py]: Yes - and can remove the `for` loop.  Here's the bottom of the function:

```python
for j in range(simple_np_arr.size):
        arr_val = arr[:, j].values
        n_arr_vals = len(arr_val)
        assert n_arr_vals <= n_elts_needed
        if n_arr_vals < n_elts_needed:
            val = np.full(shp, np.nan)
            row_idx_name = arr.dims[0]
            feature_idx = arr.indexes[row_idx_name]
            if not isinstance(feature_idx, pd.MultiIndex):
                continue
            dim_coord_pairs = feature_idx.tolist()
            for idx, dim_coord_pair in enumerate(dim_coord_pairs):
                val[dim_coord_pair] = arr_val[idx]
        else:
            val = arr_val.reshape(shp)
        layer = simple_np_arr[j]
        dset[layer] = xr.DataArray(val, coords=coords, dims=dims)
    return MLDataset(dset)
```

**Comment by gbrener** [comment]
Ok, made the changes @PeterDSteinberg .

**Comment by PeterDSteinberg** [comment]
Looks good!  Merging.

**Pull request merged**